### PR TITLE
Spectrogram log zero Issue #821, 820 & 818

### PR DESCRIPTION
**Pull request by areeda**
Note this PR includes changes in PRs 817 & 819
It addresses signals with data but have long periods of zeroes so that one of the columns in the spectrogram is all zeroes

### CODE REVIEW

File: gwpy/cli/spectrogram.py [hunk 1]
```diff
@@ -166,15 +166,21 @@ def scale_axes_from_data(self):
                 args.ymin, args.ymax)
 
         # auto scale colours
-        from numpy import percentile
+        import numpy as np
```
Comment (duncanmmacleod) [inline gwpy/cli/spectrogram.py]: @areeda, is this change necessary?

**Comment by duncanmmacleod** [comment]
@areeda, this PR seems to have the combined commits from #817 and #819 as well as anything new for this PR. Can you rebase/clean up the branch to just have the necessary commits for this patch?

**Comment by areeda** [comment]
I believe I have addressed all of the issues we talked about this morning.
 I'm not sure what to do about that codeclimate issue it seems  not like that function although I don't see it as more complex than many of the others.

**Comment by duncanmmacleod** [comment]
@areeda, this PR now includes more commits than it did before, but admittedly most of those are already merged, so they don't impact things too much. However, this does make it hard to review this PR for what it is.

Specifically, this PR still includes 815dbeaa77699aab3a85da2adf8ed8f316c1329d, which I maintain we don't need, so if that can be excised so that this PR can continue forward, that would be good.

**Comment by areeda** [comment]
This PR is being superseded.
